bin/webhook_server

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Usage:
+    webhook_server --port <port>
+
+Arguments:
+    <port> -- The port to run the server on.
+
+Options:
+    -v --logging (debug | info | error)
+"""
+
+import logging
+import BaseHTTPServer
+from docopt import docopt
+import json
+
+from gitenberg.autoupdate import __version__
+
+
+FORMAT = '%(asctime)-15s %(message)s'
+
+def setup_logging(arguments):
+    """ creates a logger with our hard-coded configuration
+    takes: a docopt arguments object instance
+    """
+    logger = logging.getLogger('')
+    logging.basicConfig(filename='./gitburg.log', level=logging.DEBUG, format=FORMAT)
+    #stdout_handler = logging.StreamHandler(sys.stdout)
+    #logger.addHandler(stdout_handler)
+
+
+    if ('--logging' or '-v') in arguments:
+        # if
+        log_level = arguments['--logging']
+        if log_level == 'debug':
+            logger.setLevel(logging.DEBUG)
+        elif log_level == 'info':
+            logger.setLevel(logging.INFO)
+        elif log_level == 'error':
+            logger.setLevel(logging.ERROR)
+
+    return logger
+
+class RequestHandler(BaseHTTPServer.BaseHTTPRequestHandler):
+  def do_POST(self):
+    if self.path != '/payload':
+      self.send_error(404)
+      return
+
+    # TODO(Brian): https://developer.github.com/webhooks/securing/
+
+    content_length = int(self.headers['Content-Length'])
+    payload_string = self.rfile.read(content_length)
+    try:
+      payload = json.loads(payload_string)
+    except ValueError:
+      self.send_error(400)
+      return
+    print('Payload is %s' % repr(payload))
+
+    self.send_response(200)
+    self.end_headers()
+
+if __name__ == '__main__':
+    arguments = docopt(__doc__, version=__version__)
+
+    logger = setup_logging(arguments)
+
+    port = int(arguments.get('<port>', '80'))
+
+    server = BaseHTTPServer.HTTPServer(('0.0.0.0', port), RequestHandler)
```

Comment:
Have you thought about how to get an SSL  certificate for this? to use a certificate for a server on EC2, you have to install the certicate in load balacer. So its a bit involved to use Let's Encrypt. https://community.letsencrypt.org/t/what-is-the-replacement-for-the-le-renewal-python-basehttpserver-script/50626/4 Another way to do it is to run the server on elastic beanstalk and use an amazon certificate the way the gitenberg server is being run.

Reply:
Nevermind- our cert for unglue.it is a wildcard, so it should be easy to install. what do you want to call the update server?

Reply:
How does `webhooks.unglue.it` sound? Seems reasonable to use this same infrastructure for any future webhooks that aren't directly related to gitenberg, so no reason to have gitenberg in the name.

Also, FWIW, [GitHub sends a hash in the payloads](https://developer.github.com/webhooks/securing/#validating-payloads-from-github). I was thinking of just relying on that, but after reading a bit more just now I realize it's sha1 which makes SSL seem like a good idea.

Reply:
webhooks.unglue.it should work.
not using SSL could cause other problems down the road, for example if some private info is passed

Reply:
How should I get that private key to the production server? Looks liker regluit uses ansible-vault, which this project otherwise doesn't need.

Reply:
if the server is done in elastic beanstalk, env variables are set in the configuration. you just want to launch a docker image, then you'd want to build it into the docker build or do it by hand.